atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/TimestampsByCellResultWithToken.java

```diff
@@ -123,6 +127,13 @@ private TimestampsByCellResultWithToken checkNextEntryAndCreateToken() {
             }
         } else {
             flushRowBuffer();
+            if (currentRow != null) {
+                byte[] nextRow = RangeRequests.getNextStartRowUnlessTerminal(reverse, currentRow);
+                if (nextRow != null) {
+                    moreResults = true;
+                    token = ImmutableToken.builder().row(nextRow).shouldSkip(false).build();
+                }
+            }
```

Comment:
We could extract this to `buildTokenIfMoreRowsAvailable` or something along those lines.